tensorboard/webapp/tbdev_upload/tbdev_upload_test.ts

```diff
@@ -27,7 +29,13 @@ describe('tbdev upload test', () => {
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [MatButtonModule, MatDialogModule, NoopAnimationsModule],
+      imports: [
```

Comment:
I spent quite a bit of time trying to figure out how to test this.

I tried seeing if the DOM has anything I could inspect: I couldn't find anything - no trace of the directive remains in the DOM after render.

I tried getting a reference to the CdkCopyToClipboard instance so I could spy on it or, alternatively, inject my own Spy.  But I could not figure out how to do this for a directive.

Any suggestions of what I might do?

Reply:
Interesting. Probably obvious but did you try creating the DialogComponent (as opposed to the ButtonComponent) and querying for the directive? `By.directive(ClipboardDirective)` (pseudo-code)

Reply:
Stephan offline pointed me to `.injector.get()`.  So it would look like this:

```
const cdkCopyToClipboard = fixture.debugElement.query(By.css('.command-copy')).injector.get(CdkCopyToClipboard)
````

But once I got my reference to CdkCopyToClipboard I realized it wasn't what I wanted.  What I actually wanted to do is get a reference to the underlying angular Clipboard object.  I've done that now and I can verify that a specific set of text has been copied to the clipboard.